manual/src/refman/extensions/arrayliterals.etex

```diff
@@ -0,0 +1,74 @@
+(Introduced in OCaml 5.4)
+
+Since OCaml 5.4, array literal syntax @"[|" e1 ';' e2 ';' ... ';' eN "|]"@ can
+be used to denote values of type "'a array" as well as "floatarray", both in
+expression and pattern positions. This syntax is also used to display "floatarray" values in the toplevel.
+
+The compiler will use contextual information to infer the type of the array
+literal. However, in the absence of any specific contextual information, it will
+be assumed that the array literal is of type "'a array":
+
+\begin{caml_example}{verbatim}
+let x = [|42.|];;
+\end{caml_example}
```

Comment:
I would explicitly say that `x` gets the type `float array` here.

Reply:
Even though it is contained in the evaluation output, I still added the sentence to the documentation; the repetition did not feel superfluous.